cmd/gateway-gcs.go

```diff
@@ -1060,20 +1060,24 @@ func (l *gcsGateway) GetBucketPolicies(bucket string) (policy.BucketAccessPolicy
 	if err != nil {
 		return policy.BucketAccessPolicy{}, gcsToObjectError(traceError(err), bucket)
 	}
-
+	isPolicySet := false
 	policyInfo := policy.BucketAccessPolicy{Version: "2012-10-17"}
 	for _, r := range rules {
 		if r.Entity != storage.AllUsers || r.Role == storage.RoleOwner {
 			continue
 		}
+		isPolicySet = true
 		switch r.Role {
 		case storage.RoleReader:
 			policyInfo.Statements = policy.SetPolicy(policyInfo.Statements, policy.BucketPolicyReadOnly, bucket, "")
 		case storage.RoleWriter:
 			policyInfo.Statements = policy.SetPolicy(policyInfo.Statements, policy.BucketPolicyWriteOnly, bucket, "")
 		}
 	}
-
+	//Return NoSuchBucketPolicy error, when policy is not set
```

Comment:
We shouldn't do it like this @kannappanr we can simply do `len(policyInfo.Statements) == 0` and return PolicyNotFound{}

Reply:
Done.

Reply:
Comment needs to have "// " space - this is Go styling.